### ISSUE TITLE
Add missing appmodule.AppModule funcs to each sdk module in ibc-go 

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc-go#4217]
## Summary

v0.50 SDK now has two interfaces:
- module.AppModule (types package)
- appmodule.AppModule (core package)

## Problem Definition

Our modules in the SDK upgrade feature branch do not implement the second interface causing autocli registration to fail.

## Proposal

Add the following functions and compiler assertions to each sdk module in ibc-go (core, transfer, 27-ica, 29-fee):

```go
import       "cosmossdk.io/core/appmodule"

var (
     _ appmodule.AppModule        = (*AppModule)(nil)
)

// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
func (am AppModule) IsOnePerModuleType() {}

// IsAppModule implements the appmodule.AppModule interface.
func (am AppModule) IsAppModule() {}
```

Remove the following lines of code after. [Here](https://github.com/cosmos/ibc-go/blob/feat/upgrade-sdk-v0.50/testing/simapp/simd/cmd/root.go#L239) and [here](https://github.com/cosmos/ibc-go/blob/feat/upgrade-sdk-v0.50/testing/simapp/simd/cmd/root.go#L265). Remove the basic manager input argument to each query/tx command function as well. 

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by colin-axner** [comment]
It might be possible to make this change to main (unsure)

**Comment by crodriguezvega** [comment]
Closed by #4332
